internal/cli/output_opts.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2020 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cli
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/mattn/go-isatty"
+	"github.com/mongodb/mongocli/internal/config"
+	"github.com/mongodb/mongocli/internal/jsonwritter"
+	"github.com/mongodb/mongocli/internal/templatewritter"
+)
+
+const (
+	jsonFormat     = "json"
+	goTemplate     = "go-template"
+	goTemplateFile = "go-template-file"
+)
+
+var templateFormats = []string{goTemplate, goTemplateFile}
+
+type OutputOpts struct {
```

Comment:
This is the new add, and now every command that wants to print should do so with the help of this struct